eZ/Publish/API/Repository/Values/Content/DraftList/Item/UnauthorizedContentDraftListItem.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace eZ\Publish\API\Repository\Values\Content\DraftList\Item;
+
+use eZ\Publish\API\Repository\Values\Content\DraftList\ContentDraftListItemInterface;
+use eZ\Publish\API\Repository\Values\Content\VersionInfo;
+
+/**
+ * Item of content drafts list which represents draft to which user has no access for.
+ */
+class UnauthorizedContentDraftListItem implements ContentDraftListItemInterface
```

Comment:
:confused: Why do we need to push this logic to API consumer?
As in:
1. drafts are personal so at least from before you always have access to YOUR drafts.
2. even if there are cases where permissions won't let you, then I would expect it to behave like other API methods, aka filter out what you don't have access to.

Reply:
Ad 1 - You can pass user as an argument to `loadContentDrafts(User $user = null)` so we have a case with a user's drafts which not belongs to you.
Ad 2 - To create "classic" pagination we need a total number of items. We can not use the search query to get it, because we have zeta components. This is a 3rd line issue and we do not have time to get rid of them. Also, it would be a BC break I think. This is why I used `UnauthorizedContentDraftListItem` here and then in Admin UI.

Reply:
> Also, it would be a BC break I think

This is new API endpoint, so there is no BC on it before it's released, that is when it will become harder to change it ;)

Reply:
> You can pass user as an argument to loadContentDrafts(User $user = null) so we have a case with a user's drafts which not belongs to you.

Good point 👍

But........ 😏  Then maybe you just made a good argument for dropping this argument on the new API method in the first place. If people want to get drafts for other users they can either still use the old one or set current user to the user they want to fetch for, it's not the primary use case for this API so we don't need to optimize for that imho.

Dropping that argument seems like it will avoid a lot of the complexity introduced here.

Reply:
The goal was to keep all functionalities including the possibility to fetch drafts of another user even if it is a big number of them. Also,  it will still be the issue when a user loses access to some of their drafts. I am against developing a feature which is buggy by design, even if it looks like a corner case. 

We have a similar problem in content reverse relations where it is a common situation when a user has no access to some contents. And we have the plan to solve this issue in the same way (return the list with two kinds of items)